### PR TITLE
DOI flow discord alerts

### PR DESCRIPTION
**Pull request by shadrach-tayo**
## Note: Add new env to k8s vaults
```
DISCORD_NOTIFICATIONS_DOI_WEBHOOK_URL=<copy webhook url from #doi-minting channel on discord>
```
## Description of the Problem / Feature
- No way to easily track DOI minting and crossref callback status

## Explanation of the solution
- Make discordNotify generic and reusable
- Send discord notifications for DOI registration and crossref callback

### CODE REVIEW

**Comment by m0ar** [comment]
@shadrach-tayo ennvar I've set the envvar for you under `app/{dev,staging,production}` :heavy_check_mark:

File: desci-server/src/utils/discordUtils.ts [hunk 1]
```diff
@@ -2,11 +2,64 @@ import { DiscordNotification } from '@penseapp/discord-notification';
...
+  | DiscordNotification['infoMessage']
+  | DiscordNotification['warningMessage']
+  | DiscordNotification['errorMessage'];
+
+export const discordNotify = async ({
+  message,
+  title = 'Node Updated',
+  type = DiscordNotifyType.SUCCESS,
+  channel = DiscordChannel.NodesFeed,
+}: {
```
Comment (m0ar) [inline desci-server/src/utils/discordUtils.ts:38]: I'd like to see this taking an `env` param as well that we bake into the discord message. Make it a bit more obvious what the source of the event is :eyes:

**Review by m0ar — APPROVED** [review]
Looks good to me, but I think we should add an `env` tag to the discord messages. :ok_hand: 

I've set the envvar for you in `app/{dev,staging,production}` ✔️

**Comment by hubsmoke** [comment]
@shadrach-tayo if DISCORD_NOTIFICATIONS_DOI_WEBHOOK_URL is not set, this seems to crash the app?

**Comment by shadrach-tayo** [comment]
> @shadrach-tayo if DISCORD_NOTIFICATIONS_DOI_WEBHOOK_URL is not set, this seems to crash the app?

No it shouldn't

**Pull request merged**